---
layout: default
title: Result Interpretation
nav_order: 4
---

# Result Interpretation Guide

For the outputs of this pipeline (`.qza` and `.qzv` files), you should be able to open them with [QIIME2 View](https://view.qiime2.org/). This page explains the purpose and interpretation of major output file generated by the 16S microbiome workflow. Most of files that are explained here are `.qzv` files, since they are visualizations of data.  

{: .note }
For more details, refer to QIIME2â€™s official [tutorial](https://amplicon-docs.qiime2.org/en/latest/). 

## ðŸ§¾ Summary Table

| Step | File(s) | Purpose | Key Insights |
|------|----------|----------|---------------|
| Demultiplexing | `demux.qzv` | Input stats & quality | Check read balance & trimming point |
| Denoising | `stats.qzv` | Filter & denoise reads | Inspect % input passed filter |
| Features | `rep-seqs.qzv`, `table.qzv` | Feature identity & abundance | Explore sequence richness |
| Phylogenetic Tree | `rooted-tree.qza`, `tree.nwk` | Evolutionary relationships | Input for diversity analysis |
| Alpha/Beta Diversity | `evenness.qzv`, `faith-pd.qzv`, `unweighted_unifrac.qzv` | Diversity comparisons | Compare diversity between groups |
| Rarefaction | `alpha-rarefaction.qzv` | Sequencing depth evaluation | Check for plateau |
| Taxonomy | `taxa-bar-plots.qzv` | Taxonomic composition | Examine dominant taxa |
| ANCOM-BC | `da-barplot.qzv`, `level-6-da-barplot-vegetation.qzv` | Differential abundance | Identify significant taxa |